content/schema-evolution-field-operations/index.md

```diff
@@ -0,0 +1,194 @@
+2. Canonicalise the item `a`: `a_c = Item.canonicalise a`.
+3. Add item `a_c` to the Register.
+
+
+### REST API Implications
+
+The REST API has no changes given that it is a read-only API. From the **API
+consumer** perspective, nothing changes but let me make it explicit here.
+
+One cornerstone assumption in the Registers' API is that a consumer is
+expected to handle graciously any Record/Item response that doesn't match
+exactly their knowledge of the schema.
+
+Scenario:
+
+Give a consumer that knows about a schema such as `Schema [("x", Number),
+("y", Number), ("z", String)]` where `x` is the primary field, such consumer
+is expected to handle reponses from the API such as:
+
+* **All known fields are present**
+
+  Given `{"x": 1, "y": 2, "z": "foo"}`, keep `{"x": 1, "y": 2, "z": "foo"}`.
+
+* **Some known fields are present**
+
+  Given `{"x": 1, "y": 2}`, keep `{"x": 1, "y": 2}` and ignore the fact that
+  `z` is missing.
+
+* **Some unknown fields present**
+
+  Given `{"x": 1, "y": 2, "z": "foo", "n": 42}`, keep `{"x": 1, "y": 2, "z": "foo"}`
+  and ignore the fact that `n` is unknown. If appropriate, update the schema
+  and handle it as the first case.
+
+
+## <span id="glossary">Glossary</span>
+
+### <span id="datatype">Datatype</span>
+
+Let `DataType` be any of the datatypes defined in the
+[specification](http://openregister.github.io/specification/#datatypes). For
+brevity, the RFC will express it as a union of strings and numbers.
+
+```elm
+type DataType = String | Number
+```
+
+### <span id="field-id">Field ID</span>
+
+Let `Id` be any valid field id, [field name in the
+specification](http://openregister.github.io/specification/#fieldname-datatype).
+
+```elm
+type Id
+```
+
+### <span id="field">Field</span>
+
+Let `Field` be a pair of `Id` and `DataType`. Although a Field has more data
+than just name and type, for the purpose of this document, we can simplify its
+definition.
+
+```elm
+type Field = (Id, DataType)
+```
+
+### <span id="schema">Schema</span>
+
+Let `Schema` be the set of fields allowed in items. A schema has a required
+primary field and optionally a set of fields (unique by `Id`). We will
+consider the first field, the primary field.
+
+```elm
+type Schema = Schema (Set Field)
+```
+
+### <span id="item">Item</span>
+
+Let `Item` be a data set conforming to the schema.
+
+```elm
+type Item = Item (Set (Id, value))
+```
+
+Where `value` is any valid value for the corresponding datatype defined in the
+schema.
+
+### <span id="schema-add-function">Schema `add` function</span>
+
+The function `Schema.add` takes a schema and a field and returns the new
+schema.
+
+```elm
+Schema.add : Schema -> Field -> Schema
+```
+
+If the field already exists the operation behaves like the identity function
+so:
+
+```elm
+schema' = Schema.add schema field -- schema' == schema
+```
+
+### <span id="item-add-function">Item `add` function</span>
+
+The function `Item.add` takes a list of items, a schema and an item and
```

Comment:
Should this be `Items.add` (plural)? I think it's meant to be the same as the example below which is plural? The title of the section also uses Item singular but would be good to have all three consistent.

Reply:
Ah yes, @karlbaker02 pointed it out as well and forgot to fix it 👍